### ISSUE TITLE
Issues with specifying vcov

### ISSUE THREAD

**Issue by NickCH-K** [s3alfisc/pyfixest#41]
1. Currently, `vcov` is a required positional argument in `feols`. Note this also means you can't specify the vcov using the `.vcov` method without also specifying it in the `feols` call as a positional argument. Suggest matching the defaults from the R package of `vcov = 'iid'` with no fixed effects specified, or clustered at the level of the first fixed effect if fixed effects are specified., or overwriting that with whatever was set in `.vcov`.
2. `help(fixest.feols)` suggests using `dict("CRV1":"clustervar")` to specify a cluster variable but this is improper syntax.
3. In my example data I'm having trouble setting any sort of clustering variable alongside a set of fixed effects (`pip install causaldata`; note there are no missing values in this data):

```python
from causaldata import restaurant_inspections
import pandas as pd
import pyfixest as pf
import numpy as np

res = restaurant_inspections.load_pandas().data
fixest = pf.Fixest(data = res)

fixest.feols('inspection_score ~ Weekend | business_name', vcov = 'iid')
# runs fine

fixest.feols('inspection_score ~ Weekend | business_name', vcov = dict({'CRV1':'business_name'}))
# TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''

# maybe it prefers numbers?
fixest.feols('inspection_score ~ Weekend | business_name', vcov = dict({'CRV1':'Year'}))
# IndexError: index 27134 is out of bounds for axis 0 with size 27076

# Or integers?
res['random_category'] = np.random.randint(0, 10, res.shape[0])
fixest = pf.Fixest(data = res)
fixest.feols('inspection_score ~ Weekend | business_name', vcov = dict({'CRV1':'random_category'}))
# IndexError: index 27087 is out of bounds for axis 0 with size 27076

# or categoricals?
res['categorical'] = pd.Categorical(res['random_category'])
fixest = pf.Fixest(data = res)
fixest.feols('inspection_score ~ Weekend | business_name', vcov = dict({'CRV1':'categorical'}))
# IndexError: index 27087 is out of bounds for axis 0 with size 27076

# Works fine without the FEs
fixest.feols('inspection_score ~ Weekend', vcov = dict({'CRV1':'categorical'}))

# although still not for business_name
fixest.feols('inspection_score ~ Weekend', vcov = dict({'CRV1':'business_name'}))
# TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''


# Maybe it can't handle single-row FEs alongside clusters?
res['counts'] = (res
		   .groupby('business_name')['business_name']
		   .transform('size'))
fixest = pf.Fixest(data = res.query('counts > 1'))
fixest.feols('inspection_score ~ Weekend | business_name', vcov = dict({'CRV1':'business_name'}))
# ValueError: matrix should have the same number of rows as fixed effect IDs.
```

**Comment by s3alfisc** [comment]
Thanks @NickCH-K - you actually stumbled over two independent errors. The type error arises because I was calling `np.isnan()` when I should have called `pd.isna()`. The other error is due to the fact that `pyhdfe` drops 102 singletons (in which case I should definitely prompt a notification), and the cluster variable is constructed based on the input data frame (including FE singletons, but minus NA values), whereas Y, X exclude the singleton fixed effects. I should also add a test with singleton fixed effects. I will fix these errors tomorrow. As I said in the other thread, this is super useful feedback, and highly appreciated!

**Comment by s3alfisc** [comment]
Fixed with version 0.3.7 =) It is also no longer required to specify a vcov type, and pyfixest now mimics fixest's default (clustering by the first fixed effects, or iid inference when no fixed effect is specified).
